dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -473,6 +481,43 @@ public Response getVersion(@PathParam("id") String datasetId, @PathParam("versio
         }));
     }
 
+    @GET
+    @Path("{id}/uningest")
+    public Response listUningestableFiles(@PathParam("id") String datasetId) {
+        return allowCors(response(req -> ok(uningestInfoService.listUningestableFiles(findDatasetOrDie(datasetId)).stream()
+                .map(UningestableItemDTO::fromDatafile)
+                .collect(Collectors.toList()))));
+    }
+
+    @POST
+    @ApiWriteOperation
+    @Path("{id}/uningest")
+    public Response uningestFiles(@PathParam("id") String datasetId, JsonObject json) {
+        return allowCors(response(req -> {
+            UningestRequestDTO rq = jsonParser().parseUningestRequest(json);
+
+            List<DataFile> dataFiles = uningestInfoService.listUningestableFiles(findDatasetOrDie(datasetId)).stream()
+                    .filter(df -> rq.getDataFileIds().isEmpty() || rq.getDataFileIds().contains(df.getId()))
+                    .collect(Collectors.toList());
+
+            List<String> uningestFailedFileIds = new ArrayList<>();
+            for(DataFile df : dataFiles) {
+                try {
+                    execCommand(new UningestFileCommand(req, df));
```

Comment:
Since `UningestFileCommand` can be executed by superuser only I am wondering if we should check if user is superuser before trying to do any uningest. That way we do not expose any id of datafiles that the potential user do not have access to.

Reply:
I agree, added a check for super user at the beginning.